pkg/repository/prisma/job_run.go

```diff
@@ -98,23 +78,52 @@ func (j *jobRunEngineRepository) RegisterWorkflowRunRunningCallback(callback rep
 }
 
 func (j *jobRunEngineRepository) SetJobRunStatusRunning(ctx context.Context, tenantId, jobRunId string) error {
-	wrId, err := setJobRunStatusRunning(ctx, j.pool, j.queries, j.l, tenantId, jobRunId)
+
+	tx, commit, rollback, err := sqlchelpers.PrepareTx(ctx, j.pool, j.l, 5000)
 
 	if err != nil {
 		return err
 	}
 
-	for _, cb := range j.wrRunningCallbacks {
-		cb.Do(j.l, tenantId, *wrId)
+	defer rollback()
+
+	err = j.SetJobRunStatusRunningWithTx(ctx, tx, tenantId, jobRunId)
+
+	if err != nil {
+		return err
+	}
+
+	err = commit(ctx)
+	if err != nil {
+		return err
+
+	}
+
+	return nil
+}
+
+func (s *sharedRepository) SetJobRunStatusRunningWithTx(ctx context.Context, tx dbsqlc.DBTX, tenantId, jobRunId string) error {
+	wrId, err := setJobRunStatusRunning(ctx, tx, s.queries, s.l, tenantId, jobRunId)
+
+	if err != nil {
+		return err
+	}
+
+	for _, cb := range s.wrRunningCallbacks {
+		cb.Do(s.l, tenantId, *wrId)
 	}
 
 	return nil
 }
 
-func (j *jobRunEngineRepository) ListJobRunsForWorkflowRun(ctx context.Context, tenantId, workflowRunId string) ([]*dbsqlc.ListJobRunsForWorkflowRunRow, error) {
+func (j *sharedRepository) ListJobRunsForWorkflowRun(ctx context.Context, tenantId, workflowRunId string) ([]*dbsqlc.ListJobRunsForWorkflowRunRow, error) {
```

Comment:
```suggestion
func (j *sharedRepository) listJobRunsForWorkflowRun(ctx context.Context, tenantId, workflowRunId string) ([]*dbsqlc.ListJobRunsForWorkflowRunRow, error) {
```

Reply:
This is called from WorkflowService so needs to be on the JobRun and exported unless I'm missing something?

`			jobRun, err := t.config.EngineRepository.JobRun().ListJobRunsForWorkflowRun(ctx.Request().Context(), tenant.ID, runIdStr)
`